tests/webview/vendorbooks/test_vendorbooks_load.py

```diff
@@ -0,0 +1,30 @@
+import requests
+from tests import markers
+from pages.webview.home import Home
+
+
+@markers.vendor
+@markers.test_case("C593141")
+@markers.nondestructive
+def test_vendor_books_not_redirecting(vendor_base_url, selenium):
+
+    home = Home(selenium, vendor_base_url).open()
+    books = []
+
+    # GIVEN the openstax list
+    for op_book in home.featured_books.openstax_list:
```

Comment:
Do you need to append all the book titles to the list? Couldn't you use a for loop and iterate over all the books and extract the title link from the page object.

```python
vendor_home = Home(selenium, vendor_base_url).open(()
for book in home.featured_books.openstax_list:
    title_link = book.title_link
    title_link.click()
    
    current_url = home.current_url
    data = requests.get(current_url)
    
    assert vendor_base_url in current_url
    assert 200 == data.status_code
```

Is there a reason something like ☝️ doesn't work?

Reply:
I tried this and other ways but the everpresent problem with this is the stale element. Karen and I tried various ways to overcome it. This is one of them. It reloads the base page after book is clicked and no stale element exception occurs.